src/main/java/com/st/utopia/cancellationservice/model/Ticket.java

```diff
@@ -0,0 +1,191 @@
+package com.st.utopia.cancellationservice.model;
+
+import java.time.LocalDateTime;
+
+import javax.persistence.Column;
+import javax.persistence.EmbeddedId;
+import javax.persistence.Entity;
+import javax.persistence.Table;
+
+/**
+ * An Ticket model.
+ *
+ * @author Henry Cho
+ * @author Al-amine AHMED MOUSSA
+ * 
+ * Model used for the purpose of the cancellation microservice.
+ * 
+ * There are no set methods for information about the ticket such as flight number, row/seat, or class.
+ * 
+ * Upon cancelling the ticket, the microservice needs to set the following fields as null:
+ * 	-reserver
+ * 	-price
+ * 	-reservation_timeout
+ * 	-booking_id
+ * 
+ * This model does not need to define a composite key since the cancellation microservice will search for the ticket via booking_id which is unique.
+ */ 
+
+@Entity
+@Table(name="tbl_tickets")
+public class Ticket{
+	/**
+	 * The three fields that together constitute the identity of a Ticket.
+	 */
+	@EmbeddedId
+	private TicketIdentity id;
+	
+	@Column(name="class")
+	private Integer tier; //class is a keyword 
+	
+	@Column(name="reserver")
+	private Integer reserver;
+	
+	@Column(name="price")
+	private Integer price;
+	
+	@Column(name="reservation_timeout")
+	private LocalDateTime reservation_timeout;
+	
+	@Column(name="booking_id")
+	private String booking_id;
+	/**
+	 * the Getters and Setters of the Class
+	 */
+	public TicketIdentity getId() {
+		return id;
+	}
+
+	public void setId(TicketIdentity id) {
```

Comment:
Do we really want to allow the service code (or any other caller) to change the identity of a ticket object? If not, don't provide a setter; Hibernate will be using its own reflection-based method of setting fields instead of calling setters anyway.

Reply:
Yes, that should be handled by our Mysql and it should not change.

Reply:
Nevermind, it is not supported by SQL